src/WebJobs.Extensions.Http/Extensions/HttpRequestExtensions.cs

```diff
@@ -56,32 +57,59 @@ public static bool IsJsonContentType(this HttpRequest request)
                 string.Equals(headerValue.MediaType, "application/json", StringComparison.OrdinalIgnoreCase);
         }
 
-        public static ClaimsIdentity GetAppServiceIdentity(this HttpRequest request)
+        public static bool TryGetAppServiceIdentity(this HttpRequest request, out ClaimsIdentity claimsIdentity)
         {
+            claimsIdentity = null;
             if (!request.Headers.ContainsKey(EasyAuthIdentityHeader))
             {
-                return null;
+                return false;
             }
+
             string headerValue = request.Headers[EasyAuthIdentityHeader].First();
-            return FromBase64EncodedJson(headerValue);
+            ClaimsIdentitySlim claimsIdentitySlim = FromBase64EncodedJson<ClaimsIdentitySlim>(headerValue, EasyAuthClaimsIdentitySerializer.Value);
+            if (claimsIdentitySlim.Equals(default(ClaimsIdentitySlim)))
+            {
+                return false;
+            }
+
+            claimsIdentity = claimsIdentitySlim.ToClaimsIdentity();
+            return true;
+        }
+
+        public static bool TryGetStaticWebAppsIdentity(this HttpRequest request, out ClaimsIdentity claimsIdentity)
+        {
+            claimsIdentity = null;
```

Comment:
This method duplicates the logic above. Can we consolidate the logic in a single method instead? We can expose a new argument (enum) so the consumer can indicate the expected identity type which would drive the logic for which serializer to use and the comparisons.

Your `FromBase64EncodedJson` can also revert to the previous behavior where it returns the claims identity. That would require you to add a constraint and introduce an interface that is implemented by the principal/claims types, defining the method they'd need to implement (`ToClaimsIdentity`). This would allow you to simplify this and avoid duplication.